Package tidy

edunham opened the pull request:
This fixes https://github.com/servo/servo/issues/861. 

@askeing, I've copied your work from https://github.com/askeing/servo_tidy and attributed the commit to you. My commit in this PR is Git housekeeping to preserve `tidy`'s history. If you'd like to make additional changes, I've given you and @shinglyu push access to my fork of Servo. Apologies if this is already familiar, but the workflow for pushing to my branch is:

```
# in your clone of the Servo repo
$ git remote add edunham git@github.com:edunham/servo.git
$ git checkout -b package-tidy
$ git pull edunham package-tidy
# make commits that you'd like to see in this PR
$ git push edunham package-tidy
```

Once this lands, I'll look at how to publish it to PyPI and automate that process.

Please don't merge this yet; we still need to discuss how the change should work around https://github.com/servo/servo/blob/master/python/servo/testing_commands.py#L33 , as I've yet to figure out how to get the egg to actually expose its tests.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10590)

<!-- Reviewable:end -->

Commit:
start Tidy move

askeing committed:
Create a Python package of Tidy

Commit:
add tidy to README

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @pcwalton (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/tidy.py, python/tidy/tests/__init__.py, python/tidy_self_test/tidy_self_test.py, python/README.md, python/tidy/HISTORY.rst, python/tidy/servo_tidy/licenseck.py, python/tidy/setup.py, python/tidy_self_test/__init__.py, python/tidy_self_test/rust_tidy.rs, python/tidy_self_test/wrong_space.rs, python/tidy/servo_tidy/__init__.py, python/tidy_self_test/long_line.rs, python/tidy/tests/test_tidy.py, python/tidy/tests/test.toml, python/tidy_self_test/whatwg_link.rs, python/tidy/tests/wrong_space.rs, python/tidy/tests/long_line.rs, python/tidy/tests/rust_tidy.rs, python/tidy/README.rst, python/tidy/tests/tidy_self_test.py, python/tidy_self_test/spec.webidl, python/tidy/tests/whatwg_link.rs, python/tidy/Makefile, python/tidy/tests/spec.webidl, python/tidy_self_test/incorrect_license.rs, python/tidy/tests/incorrect_license.rs, python/tidy_self_test/speclink.rs, python/tidy/servo_tidy/tidy.py, python/tidy/tests/speclink.rs, python/tidy_self_test/test.toml

edunham commented:
Do we know whether anyone's actually running the Tidy selftests through the mach-centric `testing_commands.py` wrapper? If that's not a priority, removing that functionality would be the easiest workaround.

jdm commented:
I'm not entirely clear what you're asking; TravisCI is running `./mach test-tidy --self-test` which corresponds to that block in `testing_commands.py`, I believe.

askeing committed:
Merge branch 'package-tidy' of github.com:edunham/servo into package-tidy

askeing committed:
Package tidy
- Modified the testing commands
- Added the requirements
- Moved python/tidy/tests to python/tidy/servo_tidy_tests for self tidy tests

askeing commented:
Due to servo_tidy cannot be found on PyPI now, the test will failed.
Here's the local test steps:
```bash
$ cd <SERVO>
$ rm -rf ./python/_virtualenv/
$ ./mach help
$ # show error msg: Could not find a version that satisfies the requirement servo-tidy==0.0.1
$ 
$ source ./python/_virtualenv/bin/activate
$ pip install -e python/tidy/
$ # install servo_tidy from local
$ deactivate
$ 
$ ./mach test-tidy --self-test
```

askeing committed:
Modify the description of setup.py

askeing committed:
Modified the servo_tidy_tests for both "./mach" and "make test"
- fix `cd python/tidy; make test` fail issue

Commit:
Use local `tidy`

https://github.com/servo/servo/issues/861#issuecomment-209547756

"I think the most important concern is that it's possible to modify tidy.py
and see how those changes affect ./mach test-tidy with the fewest possible
intermediate steps." - jdm

This takes publishing complexity away from the contributor when testing
changes and makes it an infra problem instead, where it's much easier to
automate & saner to manage pypi credentials

askeing commented on python/requirements.txt:
```diff
@@ -14,4 +14,4 @@ pyflakes == 0.8.0
 ply == 3.8
 
-servo_tidy == 0.0.1
+-e python/tidy
```
Cool!
I never knew we can use `-e foo/bar` in requiremnets.txt before.
It's really helpful to avoid the publish problems, and make testing easier :)

askeing commented on python/requirements.txt:
```diff
@@ -13,3 +13,5 @@ pyflakes == 0.8.0
 # For test-webidl
 ply == 3.8
+
+-e python/tidy
```
Cool!
I never knew we can use `-e foo/bar` in requiremnets.txt before.
It's really helpful to avoid the publish problems, and make testing easier :)

askeing committed:
Add servo_tidy_tests into ignore dirs, and modify setup.py to fit the tidy check

- fix the issue of '$ ./mach test-tidy --no-progress'

askeing commented:
All checks have passed :)
Do I have to rebase the commits before review? Or after get the r+?

edunham commented:
@askeing we'll only need to rebase if the master branch develops a merge conflict with these changes before the PR lands. Since there've been no major changes in `tidy` or the other files touched by this PR since the `package-tidy` branch was created, we don't have to worry about rebasing. Our next steps are to get the code reviewed, wait for bors-servo to run the tests one more time after the r+, and then watch it automatically land if nobody has landed anything conflicting in the meantime!

@bors-servo r? @larsbergstrom

larsbergstrom commented:
Reviewed 1 of 43 files at r1, 1 of 29 files at r2, 10 of 42 files at r4, 28 of 33 files at r5, 2 of 3 files at r6, 1 of 1 files at r7, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10590)*
<!-- Sent from Reviewable.io -->

larsbergstrom commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit ee433a0 has been approved by `larsbergstrom`

<!-- @bors-servo r=larsbergstrom ee433a030c306990f14a0dbbf91f1d4bf5e3003c -->

bors-servo commented:
:hourglass: Testing commit ee433a0 with merge bfe5453...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/300), [arm32](http://build.servo.org/builders/arm32/builds/320), [arm64](http://build.servo.org/builders/arm64/builds/312), [linux-dev](http://build.servo.org/builders/linux-dev/builds/318), [linux-rel](http://build.servo.org/builders/linux-rel/builds/351), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/323), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/315), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/320)
